### PR TITLE
Register container

### PR DESCRIPTION
**Pull request by qleseid**

### CODE REVIEW

File: backend/app.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// If there is a connection error with db
+db.on('error', console.error.bind(console, 'connection error:'));
+
+// If DB is opened successfully
+db.once('open', () => {
+  console.log('Connection Successful!');
+});
+
+// CORS Middleware
+app.use(cors());
```
Comment (closs290) [inline backend/app.ts:31]: I'm curious what CORS does

File: backend/routes/user.routes.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { auth } from '../middleware/auth';
+
+export const user3Routes = express.Router();
+
+user3Routes.post('/register', async (req: any, res: any) => {
+  try {
+    const user = new User(req.body);
+    await user.save();
+    const token = await user.generateAuthToken();
+    res.status(201).send({user, token});
```
Comment (closs290) [inline backend/routes/user.routes.ts:12]: ooops. I had been being lazy and just respond with '200' on success, hahaha

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -1,8 +1,15 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { LoginComponent } from './login/login.component';
+import { RegisterComponent } from './register/register.component';
 
-
-const routes: Routes = [];
+// This routes the app from specific address bar inputs
+const routes: Routes = [
```
Comment (closs290) [inline src/app/app-routing.module.ts:7]: This is awesome! Angular makes this so easy!

File: src/app/register/register.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(RegisterComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (closs290) [inline src/app/register/register.component.spec.ts:22]: Should we require unit tests before we allow our colleagues merge/submit code? Or should unit tests be their own pull requests/branches or something?

**Review by closs290 — APPROVED** [review]
Nice new functionality! I think it's time for the rest of us to start adding features!

**Review by closs290 — APPROVED** [review]

File: backend/app.ts (hunk 1 above)
Comment (nicholasjuncos) [reply backend/app.ts:31]: Cross-Origin Resource Sharing (CORS). Used to send additional headers in requests.

File: backend/routes/user.routes.ts (hunk 2 above)
Comment (nicholasjuncos) [reply backend/routes/user.routes.ts:12]: It's important to send back at least the token for authentication. Angular will take this token, and assign it to local storage. This token state you are logged in and determines your identity, and even your device and browser identity at that.

File: src/app/register/register.component.spec.ts (hunk 4 above)
Comment (nicholasjuncos) [reply src/app/register/register.component.spec.ts:22]: We definitely can. This branch does not have anything to test, although we can add it to other branches.

File: src/app/app-routing.module.ts (hunk 3 above)
Comment (qleseid) [reply src/app/app-routing.module.ts:7]: They do once you know where to find it and how to use it... :)

File: src/app/register/register.component.spec.ts (hunk 4 above)
Comment (qleseid) [reply src/app/register/register.component.spec.ts:22]: My 2 cents: our app is pretty simple and in my mind if anything fails it will render the app non-functional; unit testing through use if you will.

**Review by michaelbilliot — APPROVED** [review]
